src/libopensc/sc-ossl-compat.h

```diff
@@ -114,208 +55,11 @@ extern "C" {
 #define EVP_PKEY_get_raw_public_key(p, pu, l)   (0)
 #endif
 
+/* OpenSSL 1.1.1 has FIPS_mode function */
 #if OPENSSL_VERSION_NUMBER >= 0x30000000L
-#define EC_POINT_get_affine_coordinates_GFp     EC_POINT_get_affine_coordinates
-#define EC_POINT_set_affine_coordinates_GFp     EC_POINT_set_affine_coordinates
-
-# ifndef FIPS_mode
 #define FIPS_mode()                             EVP_default_properties_is_fips_enabled(NULL)
-# endif
-
-/* As defined in openssl/include/openssl/evp.h */
-# ifndef EVP_PK_RSA
-#  define EVP_PK_RSA      0x0001
-#  define EVP_PK_DSA      0x0002
-#  define EVP_PK_DH       0x0004
-#  define EVP_PK_EC       0x0008
-#  define EVP_PKT_SIGN    0x0010
-#  define EVP_PKT_ENC     0x0020
-#  define EVP_PKT_EXCH    0x0040
-#  define EVP_PKS_RSA     0x0100
-#  define EVP_PKS_DSA     0x0200
-#  define EVP_PKS_EC      0x0400
-# endif
-#endif
-
-#if OPENSSL_VERSION_NUMBER < 0x30000000L
-#define EVP_PKEY_eq                             EVP_PKEY_cmp
-#define EVP_PKEY_CTX_set1_rsa_keygen_pubexp     EVP_PKEY_CTX_set_rsa_keygen_pubexp
```

Comment:
This function is deprecated in OpenSSL 3.0 so we need to keep this define.

Reply:
Code was changed  EVP_PKEY_eq  to   EVP_PKEY_cmp and 
EVP_PKEY_CTX_set1_rsa_keygen_pubexp    to  EVP_PKEY_CTX_set_rsa_keygen_pubexp